infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -588,12 +605,43 @@ public void resetServer(@Nonnull RequestMsg req,
                 " sequencerEpoch = {}", globalLogTail, streamTailToGlobalTailMap.size(),
                 streamTailToGlobalTailMap, sequencerEpoch);
 
+        if (serverContext.isDeleteInactiveStreamsOnSequencerReset()) {
+            CompletableFuture<Set<UUID>> getInactiveStreamsFuture = CompletableFuture.supplyAsync(() ->
```

Comment:
Wouldn't this introduce a race condition where the sequencer is bootstrapped (allow for other clients to be serviced) but the async task is still performing a clean up on the metatdata?

Reply:
Yes. There will be a very short window after the sequencer bootstrap. But there will not be any correctness issue since a dropped table has a CLEAR update. Today there is no way to prevent the sequencer from serving requests once bootstrap is done. We will need to have two-phase sequencer bootstrap and client-aware sequencer request control to achieve that.